File: backend/Domain/Entities/PropertyPhoto.cs
using System.ComponentModel.DataAnnotations.Schema;
using Domain.Common;

namespace Domain.Entities
{
    public class PropertyPhoto : BaseEntity
    {
        [ForeignKey("Property")]
        public Guid PropertyId { get; set; }
        public required string Url { get; set; }
        
        public required Property Property { get; set; }
    }
}

File: mobile/lib/src/features/dashboard/domain/entities/property_photo.dart
import 'package:equatable/equatable.dart';

class PropertyPhoto extends Equatable {
  final String id;
  final String url;

  const PropertyPhoto({
    required this.id,
    required this.url,
  });

  @override
  List<Object?> get props => [id, url];
}
